### PR TITLE
More grapple addons

### PR DESCRIPTION
**Pull request by rplazevedo**
- Fixed being able to swing from coin
- Reset grapple on player reset/death
- Added sorting layers so Grapple doens't show in front of player

### CODE REVIEW

**Review by DanielRussom — CHANGES_REQUESTED** [review]

File: Assets/Scripts/DeathPlane.cs [hunk 1]
```diff
@@ -16,5 +16,6 @@ void OnTriggerEnter2D(Collider2D other)
     void PlayerDeath()
     {
         player.Reset();
+        player.DetachGrapple();
```
[the remark this answers is not in the document]
Comment (rplazevedo) [reply Assets/Scripts/DeathPlane.cs:19]: Fixed

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -48,10 +48,6 @@ private void Awake()
     private void SetupLineRenderer()
     {
         lineRenderer = GetComponent<LineRenderer>();
-        lineRenderer.startColor = Color.black;
-        lineRenderer.endColor = Color.black;
-        lineRenderer.startWidth = 0.02f;
-        lineRenderer.endWidth = .02f;
         lineRenderer.positionCount = 2;
         lineRenderer.material = new Material(Shader.Find("Sprites/Default"));
```
[the remark this answers is not in the document]
Comment (rplazevedo) [reply Assets/Scripts/Player.cs]: Done. We do everything in the editor now.

File: Assets/Scripts/Player.cs [hunk 3]
```diff
@@ -205,9 +198,14 @@ private void DetachGrappleOnClick()
     {
         if (Input.GetMouseButtonDown(0))
```
[the remark this answers is not in the document]
Comment (rplazevedo) [reply Assets/Scripts/Player.cs]: yes

File: Assets/Scripts/Player.cs [hunk 4]
```diff
@@ -147,7 +140,7 @@ private void FireGrapple()
     {   
         var mouseCoord = cam.ScreenToWorldPoint(Input.mousePosition);
         RaycastHit2D hit = Physics2D.Linecast(transform.position, (mouseCoord - transform.position) * maxRange);
-        if (hit)
+        if (hit && hit.collider.gameObject.layer == 6) // This is not very explicit that layer 6 is the Ground layer
```
[the remark this answers is not in the document]
Comment (DanielRussom) [reply Assets/Scripts/Player.cs:129]: Raised a ticket for this.

**Review by DanielRussom — APPROVED** [review]

**Pull request merged**